### PR TITLE
Watch later

### PR DESCRIPTION
**Pull request by darshandamre**
User can add videos to their watch later list
User can navigate to the watch later page from the sidebar
User can remove videos from watch later list from the watch later page
[Preview Deploy](https://video-library-git-watch-later-darshandamre.vercel.app/)
PS: other pages don't work as of now
image
image

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,11 +1,19 @@
 import { Route, Routes } from "react-router-dom";
+import { Layout } from "./components";
+import { RequireAuth } from "./components/RequireAuth";
```
Comment (ravikumar0403) [inline src/App.js]: ig you missed the barrel import here

File: src/App.js [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
 import { Home, Login, SignUp } from "./pages";
+import { WatchLater } from "./pages/WatchLater/WatchLater";
```
Comment (ravikumar0403) [inline src/App.js]: this can also be imported in previous line itself.

File: src/axios.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import axios from "axios";
+import { getToken } from "./utils/token";
+
+const privateReq = axios.create({
```
Comment (ivinayakg) [inline src/axios.js:8]: nice, i really like you creating a default instance to use axios

File: src/hooks/useOnClickOutside.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      if (!ref.current || ref.current.contains(event.target)) return;
+
+      handler(event);
+    };
+    if (shouldAddListener) {
+      document.addEventListener("mousedown", listener);
+      document.addEventListener("touchstart", listener);
+    }
+    return () => {
+      document.removeEventListener("mousedown", listener);
```
Comment (ivinayakg) [inline src/hooks/useOnClickOutside.js:17]: great for add a custom for creating dom events and also cleaning them up

**Review by ivinayakg — APPROVED** [review]
great work, really liked it 👍

**Review by ravikumar0403 — APPROVED** [review]
Liked the implementation of axios instance
Good use of custom hooks

File: src/App.js (hunk 1 above)
Comment (darshandamre) [reply src/App.js]: fixed

File: src/App.js (hunk 2 above)
Comment (darshandamre) [reply src/App.js]: fixed

**Pull request merged**